src/landlab/components/river_flow_dynamics/river_flow_dynamics.py

```diff
@@ -0,0 +1,1937 @@
+...         0.446,
+...         0.445,
+...         0.443,
+...         0.442,
+...         0.441,
+...         0.439,
+...         0.438,
+...         0.437,
+...         0.435,
+...         0.434,
+...         0.433,
+...         0.431,
+...         0.43,
+...         0.428,
+...         0.427,
+...         0.425,
+...         0.424,
+...         0.422,
+...         0.421,
+...         0.419,
+...         0.418,
+...         0.416,
+...         0.415,
+...         0.413,
+...         0.412,
+...         0.41,
+...         0.409,
+...         0.407,
+...         0.405,
+...         0.404,
+...         0.402,
+...         0.401,
+...         0.399,
+...         0.397,
+...         0.396,
+...         0.394,
+...         0.393,
+...         0.391,
+...         0.389,
+...         0.388,
+...         0.386,
+...         0.384,
+...         0.383,
+...         0.381,
+...         0.379,
+...         0.378,
+...     ]
+... )
+
+The calculated flow_depth is:
+
+>>> flow_depth = np.reshape(grid["node"]["surface_water__depth"], (nRows, nCols))[10, :]
+
+The average (absolute) difference between predited and expected in percentage is:
+
+>>> np.round(np.abs(np.mean(flow_depth_expected - flow_depth)) * 100, 0)
+0.0
+
+And the velocity at links along the center of the channel.
+
+The expected flow velocity is:
+
+>>> flow_velocity_expected = np.array(
+...     [
+...         0.45,
+...         0.595,
+...         0.694,
+...         0.754,
+...         0.795,
+...         0.821,
+...         0.838,
+...         0.848,
+...         0.855,
+...         0.858,
+...         0.86,
+...         0.86,
+...         0.858,
+...         0.857,
+...         0.858,
+...         0.86,
+...         0.864,
+...         0.866,
+...         0.866,
+...         0.866,
+...         0.866,
+...         0.867,
+...         0.869,
+...         0.872,
+...         0.874,
+...         0.875,
+...         0.876,
+...         0.878,
+...         0.88,
+...         0.881,
+...         0.882,
+...         0.884,
+...         0.885,
+...         0.886,
+...         0.888,
+...         0.889,
+...         0.89,
+...         0.892,
+...         0.893,
+...         0.894,
+...         0.896,
+...         0.898,
+...         0.898,
+...         0.901,
+...         0.901,
+...         0.9,
+...         0.904,
+...         0.906,
+...         0.902,
+...         0.904,
+...         0.91,
+...         0.907,
+...         0.904,
+...         0.911,
+...         0.911,
+...         0.907,
+...         0.909,
+...         0.913,
+...         0.914,
+...     ]
+... )
+
+The calculated flow velocity is:
+
+>>> linksAtCenter = grid.links_at_node[np.array(np.arange(600, 660))][:-1, 0]
+>>> flow_velocity = grid["link"]["surface_water__velocity"][linksAtCenter]
+>>> flow_velocity = np.round(flow_velocity, 3)
+
+The average (absolute) difference between predited and expected
+flow velocity in percentage is:
+
+>>> np.round(np.abs(np.mean(flow_velocity_expected - flow_velocity)) * 100, 0)
+0.0
+"""
```

Comment:
@angelmons doctests are primarily intended as documentation, which is also tested, but not as a real test of your component. The example you provide here is probably better suited as an example in a Jupyter notebook. Would it be possible to simplify your example for this doctest. Perhaps using a small grid of, maybe, 5 rows and columns? It should demonstrate to a user how to use your model and what it is doing.